usaspending_api/awards/v2/filters/matview_filters.py

```diff
@@ -74,54 +74,54 @@ def matview_search_filter(filters, model):
 
         elif key == "agencies":
             # TODO: Make function to match agencies in award filter throwing dupe error
-            funding_toptier = []
-            funding_subtier = []
-            awarding_toptier = []
-            awarding_subtier = []
+            funding_toptier = Q()
+            funding_subtier = Q()
+            awarding_toptier = Q()
+            awarding_subtier = Q()
             for v in value:
                 type = v["type"]
                 tier = v["tier"]
                 name = v["name"]
                 if type == "funding":
                     if tier == "toptier":
-                        funding_toptier.append(name)
+                        funding_toptier |= Q(toptier_agency_id__name=name)
```

Comment:
Should this be `funding_toptier_agency_name`? I see the double underscore, but I did not see the relationship setup in corresponding matview model.

Reply:
Similar question for other changes in this file below.